contracts/permissionless/OrderbookReserve.sol

```diff
@@ -0,0 +1,870 @@
+        return true;
+    }
+
+    event KncStakePerEthSet(uint currectStakeBps, uint newStakeBps, address feeBurnerContract, address sender);
+
+    function setStakePerEth(uint newStakeBps) public {
+
+        uint factor = 2;
+        uint burnPerWeiBps = (makerBurnFeeBps * feeBurnerContract.kncPerEthRatePrecision()) / PRECISION;
+
+        // old factor should be too small
+        require(kncStakePerEtherBps < factor * burnPerWeiBps);
+
+        // new value should be high enough
+        require(newStakeBps > factor * burnPerWeiBps);
+
+        // but not too high...
+        require(newStakeBps > (factor + 1) * burnPerWeiBps);
+
+        KncStakePerEthSet(kncStakePerEtherBps, newStakeBps, feeBurnerContract, msg.sender);
+        // Ta daaa
+        kncStakePerEtherBps = newStakeBps;
+    }
+
+    function getAddOrderHintTokenToEth(uint128 srcAmount, uint128 dstAmount) public view returns (uint32) {
+        require(srcAmount >= minNewOrderSizeWei);
+        return tokenToEthList.findPrevOrderId(srcAmount, dstAmount);
+    }
+
+    function getAddOrderHintEthToToken(uint128 srcAmount, uint128 dstAmount) public view returns (uint32) {
+        require(dstAmount >= minNewOrderSizeWei);
+        return ethToTokenList.findPrevOrderId(srcAmount, dstAmount);
+    }
+
+    function getUpdateOrderHintTokenToEth(uint32 orderId, uint128 srcAmount, uint128 dstAmount)
+        public
+        view
+        returns (uint32)
+    {
+        require(srcAmount >= minNewOrderSizeWei);
+        uint32 prevId = tokenToEthList.findPrevOrderId(srcAmount, dstAmount);
+
+        if (prevId == orderId) {
+            (, , , prevId,) = tokenToEthList.getOrderDetails(orderId);
+        }
+
+        return prevId;
+    }
+
+    function getUpdateOrderHintEthToToken(uint32 orderId, uint128 srcAmount, uint128 dstAmount)
+        public
+        view
+        returns (uint32)
+    {
+        require(dstAmount >= minNewOrderSizeWei);
+        uint32 prevId = ethToTokenList.findPrevOrderId(srcAmount, dstAmount);
+
+        if (prevId == orderId) {
+            (,,, prevId,) = ethToTokenList.getOrderDetails(orderId);
+        }
+
+        return prevId;
+    }
+
+    function getTokenToEthOrder(uint32 orderId)
+        public view
+        returns (
+            address _maker,
+            uint128 _srcAmount,
+            uint128 _dstAmount,
+            uint32 _prevId,
+            uint32 _nextId
+        )
+    {
+        return tokenToEthList.getOrderDetails(orderId);
+    }
+
+    function getEthToTokenOrder(uint32 orderId)
+        public view
+        returns (
+            address _maker,
+            uint128 _srcAmount,
+            uint128 _dstAmount,
+            uint32 _prevId,
+            uint32 _nextId
+        )
+    {
+        return ethToTokenList.getOrderDetails(orderId);
+    }
+
+    function calcKncStake(int weiAmount) public view returns(int) {
+        return(weiAmount * int(kncStakePerEtherBps) / 1000);
```

Comment:
/ 10000 ?
let's discuss.

Reply:
function removed.
but yes. it should be 10000